lib/lz4.h

```diff
@@ -311,10 +311,12 @@ LZ4LIB_API int LZ4_decompress_safe_partial (const char* src, char* dst, int srcS
 ***********************************************/
 typedef union LZ4_stream_u LZ4_stream_t;  /* incomplete type (defined later) */
 
+#if !defined(RC_INVOKED) /* https://docs.microsoft.com/en-us/windows/win32/menurc/predefined-macros */
```

Comment:
I don't understand the relation between `RC_INVOKED` and following declarations.
Could you explain ?

Reply:
(1) Truncation causes the warning "RC4011: identifier truncated"

`rc.exe` reports the following warnings

```
lz4\build\VS2022\..\..\lib\lz4.h(314): warning RC4011: identifier truncated to 'LZ4_STATIC_LINKING_ONLY_DISABLE'
lz4\build\VS2022\..\..\lib\lz4.h(401): warning RC4011: identifier truncated to 'LZ4_STATIC_LINKING_ONLY_DISABLE'
```

The reason of the warning is trunction.  Since `rc` is a legacy resource compiler, it truncates long symbols.


(2) `rc.exe` predefines `RC_INVOKED`

https://docs.microsoft.com/en-us/windows/win32/menurc/predefined-macros

`rc.exe` predefines `RC_INVOKED`.  And we can use it to determine that header is being processed by `rc.exe` or not.


(3) Use `RC_INVOKED` to avoid truncation warnings.

The following `#if !defined()` block means "skip this block when `rc.exe` is trying to read this block."

```
#if !defined(RC_INVOKED)
...
#endif
```

I surround `LZ4_STATIC_LINKING_ONLY_DISABLE` with this block to avoid the warning.

----

Other possible solution may be surrounding entire header except version number with `#if !defined(RC_INVOKED)`.

Reply:
It's a good explanation.
I believe it does not transpire from the provided link only.
Could you add a word about the reason of this presence in code documentation ?